src/mol-model/structure/structure/unit/bonds/inter-compute.ts

```diff
@@ -206,6 +207,68 @@ function findPairBonds(unitA: Unit.Atomic, unitB: Unit.Atomic, props: BondComput
     builder.finishUnitPair();
 }
 
+function canAddFromIndexPairBonds(structure: Structure) {
+    for (const m of structure.models) {
+        const indexPairs = IndexPairBonds.Provider.get(m);
+        if (!indexPairs?.hasOperators) return false;
+    }
+    for (const u of structure.units) {
+        if (u.conformation.operator.key === -1) return false;
+    }
+    return true;
+}
+
+function addIndexPairBonds(structure: Structure, builder: InterUnitGraph.Builder<number, StructureElement.UnitIndex, InterUnitEdgeProps>) {
+    const opUnits = new Map<number, Set<Unit>>();
+    for (const u of structure.units) {
+        const { key } = u.conformation.operator;
+        if (opUnits.has(key)) opUnits.get(key)!.add(u);
+        else opUnits.set(key, new Set([u]));
+    }
+
+    for (const m of structure.models) {
+        const indexPairs = IndexPairBonds.Provider.get(m)!;
+        const { a, b } = indexPairs.bonds;
+        const { order, flag, key, operatorA, operatorB } = indexPairs.bonds.edgeProps;
+
+        const pairs = new Map<number, Set<number>>();
+        for (let i = 0, il = operatorA.length; i < il; ++i) {
+            const unitsA = opUnits.get(operatorA[i])!;
+            const unitsB = opUnits.get(operatorB[i])!;
+            for (const uA of unitsA) {
+                for (const uB of unitsB) {
+                    if (uA === uB || !Unit.isAtomic(uA) || !Unit.isAtomic(uB)) continue;
+
+                    const h = cantorPairing(uA.id, uB.id);
+                    if (pairs.has(h)) pairs.get(h)!.add(i);
+                    else pairs.set(h, new Set([i]));
+                }
+            }
+        }
+
+        const { invertedIndex } = Model.getInvertedAtomSourceIndex(m);
+        const unitIds: [number, number] = [-1, -1];
+        pairs.forEach((indices, h) => {
+            const [unitIdA, unitIdB] = invertCantorPairing(unitIds, h);
```

Comment:
Optionally, to not rely on the pairing inversion, it should work to use `new Map<number, [unitA: number, unitB: number, indices: Set<number>]>()` instead for the `pairs`.